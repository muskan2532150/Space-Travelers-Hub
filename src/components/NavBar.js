import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import planetImage from '../assets/planet.png';

const NavBar = () => (
  <header className="header">
    <nav className="nav-bar">
      <Link to="/rocket" className="logo">
        <img src={planetImage} alt="planet" />
        <h1>Space Traveler&#39;s Hub</h1>
      </Link>
      <div className="nav-links">
        <NavLink
          to="/rocket"
          className={({ isActive }) => (isActive ? 'active' : '')}>
          Rockets
        </NavLink>
        <NavLink
          to="/missions"
          className={({ isActive }) => (isActive ? 'active' : '')}>
          Missions
        </NavLink>
        <NavLink
          to="/profile"
          className={({ isActive }) => (isActive ? 'active' : '')}>
          My Profile
        </NavLink>
      </div>
    </nav>
  </header>
);

export default NavBar;
