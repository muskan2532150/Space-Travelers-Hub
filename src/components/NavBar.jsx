import React from "react";
import { NavLink, Link } from "react-router-dom";
import planetImage from "../assets/planet.png";

const NavBar = () => {
  return (
    <header className="header">
      <nav className="nav-bar">
        <Link to="/" className="logo">
          <img src={planetImage} alt="planet" />
          <h1>Space Traveler's Hub</h1>
        </Link>

        <div className="nav-links">
          <NavLink to="/rocket" className={({isActive}) => (isActive ? "active" : "")}>Rockets</NavLink>
          <NavLink to="missions" className={({isActive}) => (isActive ? "active" : "")}>Missions</NavLink>
          <NavLink to="profile" className={({isActive}) => (isActive ? "active" : "")}>My Profile</NavLink>
        </div>
      </nav>
    </header>
  );
};

export default NavBar;
