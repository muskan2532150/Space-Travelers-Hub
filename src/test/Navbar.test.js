import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import NavBar from '../components/NavBar';

test('Demo test', () => {
  expect((1 + 2)).toBe(3);
});

test('renders Navbar Component', () => {
  const navbar = render(
    <BrowserRouter>
      {' '}
      <NavBar />
    </BrowserRouter>,
  );
  expect(navbar).toMatchSnapshot();
});
