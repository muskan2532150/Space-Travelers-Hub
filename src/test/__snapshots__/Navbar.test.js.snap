// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
       
      <header
        class="header"
      >
        <nav
          class="nav-bar"
        >
          <a
            class="logo"
            href="/"
          >
            <img
              alt="planet"
              src="planet.png"
            />
            <h1>
              Space Traveler's Hub
            </h1>
          </a>
          <div
            class="nav-links"
          >
            <a
              class=""
              href="/rocket"
            >
              Rockets
            </a>
            <a
              class=""
              href="/missions"
            >
              Missions
            </a>
            <a
              class=""
              href="/profile"
            >
              My Profile
            </a>
          </div>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
     
    <header
      class="header"
    >
      <nav
        class="nav-bar"
      >
        <a
          class="logo"
          href="/"
        >
          <img
            alt="planet"
            src="planet.png"
          />
          <h1>
            Space Traveler's Hub
          </h1>
        </a>
        <div
          class="nav-links"
        >
          <a
            class=""
            href="/rocket"
          >
            Rockets
          </a>
          <a
            class=""
            href="/missions"
          >
            Missions
          </a>
          <a
            class=""
            href="/profile"
          >
            My Profile
          </a>
        </div>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
